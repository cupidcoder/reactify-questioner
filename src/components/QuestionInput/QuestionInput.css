.QuestionInput {
  width: 90%;
  margin: 2rem auto;
  border-bottom: 1px solid rgba(46,62,72,.12);
  padding-bottom: 1rem;
}

.InputGroup {
  width: 90%;
  margin: 1rem auto;
}

.Input {
  width: 100%;
  border: 1px solid #000000;
  padding: 2rem;
  font-size: 1.2rem;
  margin-bottom: 1rem;
  display: block;
}
