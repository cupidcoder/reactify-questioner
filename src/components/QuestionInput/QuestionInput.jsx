import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { postQuestionRequest } from '../../actions/postQuestionActions';
import { getMeetupQuestionsRequest } from '../../actions/getMeetupQuestionsActions';
import style from './QuestionInput.css';
import Button from '../Button/Button';
import Spinner from '../Spinner/Spinner';

class QuestionInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      question: ''
    };
  }

  refreshQuestionsList = async () => {
    const { meetupId, refreshQuestions } = this.props;
    await refreshQuestions(meetupId);
    this.setState({ question: '' });
  }

  handleInputChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  }

  handleFormSubmit = async (e) => {
    e.preventDefault();
    const { question } = this.state;
    const { meetupId, postQuestion } = this.props;
    const questionObject = {
      meetupID: meetupId,
      body: question
    };
    await postQuestion(questionObject);
    const { status, message } = this.props;
    if (status === 201) {
      toast(message, {
        type: 'success',
        onClose: this.refreshQuestionsList
      });
    } else {
      toast(message, {
        type: 'error',
      });
    }
  }

  render() {
    const { question } = this.state;
    const { isLoading } = this.props;
    const spinner = isLoading ? <Spinner /> : null;
    return (
      <div className={style.QuestionInput}>
        <form onSubmit={this.handleFormSubmit}>
          <div className={style.InputGroup}>
            <textarea
              className={style.Input}
              placeholder='Ask your question'
              id='question'
              onChange={this.handleInputChange}
              value={question}
              required
            />
            <Button>
              Post
              {' '}
              {spinner}
            </Button>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.postQuestion.isLoading,
  status: state.postQuestion.status,
  message: state.postQuestion.message
});

const mapDispatchToProps = dispatch => ({
  postQuestion: questionObject => dispatch(postQuestionRequest(questionObject)),
  refreshQuestions: meetupId => dispatch(getMeetupQuestionsRequest(meetupId))
});

const QuestionInputComponent = connect(mapStateToProps, mapDispatchToProps)(QuestionInput);

export {
  QuestionInput,
  QuestionInputComponent
};
