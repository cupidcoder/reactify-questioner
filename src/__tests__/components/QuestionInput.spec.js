import React from 'react';
import { shallow } from 'enzyme';
import QuestionInput from '../../components/QuestionInput/QuestionInput';

describe('QuestionInput', () => {
  it('Should correctly save snapshot of the QuestionInput', () => {
    const wrapper = shallow(<QuestionInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
