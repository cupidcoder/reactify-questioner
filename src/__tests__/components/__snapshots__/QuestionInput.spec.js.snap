// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionInput Should correctly save snapshot of the QuestionInput 1`] = `ShallowWrapper {}`;
